lib/cog/command/pipeline/executor.ex

```diff
@@ -677,35 +663,160 @@ defmodule Cog.Command.Pipeline.Executor do
+    # a bit more information to show
+    command_text = case failing do
+                     %Ast.Invocation{} ->
+                     """
+                     The pipeline failed planning the invocation:
+
+                         `#{failing}`
+                     """
+                     %Plan{} ->
+                     """
+                     The pipeline failed executing the command:
+
+                         `#{failing.invocation_text}`
+
+                     """
+                     nil ->
+                       nil
+                   end
+
+    # Finally, the text of the specific error that occurred
+    error_text = """
+    The specific error was:
+
+        #{error_message}
+
+    """
+
+    [base, command_text, error_text]
+    |> Enum.reject(&is_nil/1)
+    |> Enum.join("\n")
   end
 
-  defp send_denied(which, why, state),
-    do: send_reply("Sorry, you aren't allowed to execute '#{which}' :(\n #{why}", state)
+  # Turn an error tuple into a textual message intended for chat users
+  defp format_user_error({:parse_error, msg}, _state) when is_binary(msg),
+    do: "Whoops! An error occurred. " <> msg
+  defp format_user_error({:redirect_error, invalid}, _state),
+    do: "Whoops! An error occurred. " <> redirection_error_message(invalid)
+  defp format_user_error({:binding_error, {:missing_key, var}}, _state),
+    do: "I can't find the variable '$#{var}'."
+  defp format_user_error({:binding_error, msg}, _state) when is_binary(msg),
+    do: "Whoops! An error occurred. " <> msg
+  defp format_user_error({:no_rule, current_invocation}, _state),
+    do: "No rules match the supplied invocation of '#{current_invocation}'. Check your args and options, then confirm that the proper rules are in place."
+  defp format_user_error({:denied, %Piper.Permissions.Ast.Rule{}=rule}, %__MODULE__{invocations: [current_invocation|_]}),
+    do: "Sorry, you aren't allowed to execute '#{current_invocation}' :(\n You will need the '#{rule.permission_selector.perms.value}' permission to run this command."
+  defp format_user_error({:no_relays, %Cog.Models.Bundle{name: name}}, _state),
+    do: "Whoops! An error occurred. No Cog Relays supporting the `#{name}` bundle are currently online"
+  defp format_user_error({:disabled_bundle, %Cog.Models.Bundle{name: name}}, _state),
+    do: "Whoops! An error occurred. The #{name} bundle is currently disabled"
+  defp format_user_error({:timeout, %Cog.Models.Command{}=command}, _state) do
+    name = Cog.Models.Command.full_name(command)
+    "The #{name} command timed out"
+  end
+  defp format_user_error({:template_rendering_error, {error, template, adapter}}, _state),
+    do: "Whoops! An error occurred. There was an error rendering the template '#{template}' for the adapter '#{adapter}': #{inspect error}"
+  defp format_user_error({:command_error, response}, _state),
+    do: "Whoops! An error occurred. " <> (response.status_message || response.body["message"])
```

Comment:
:+1: Having all these error messages together will help us keep the wording consistent as well.